packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -321,6 +321,15 @@ export class BlockList extends Component {
 			gridProperties,
 		} = this.props;
 		const { blockWidth } = this.state;
+
+		// Extracting the grid item properties here to avoid
+		// re-renders in the blockListItem component.
+		const isGridItem = !! gridProperties;
+		const gridItemProps = gridProperties && {
+			numOfColumns: gridProperties.numColumns,
+			tileCount: blockClientIds.length,
+			index: blockClientIds.indexOf( clientId ),
```

Comment:
Nitpick: As we named the count as `tileCount`, I'm wondering if we should also use the `tile` word here and name it `tileIndex`, wdyt? I'd like to note that I don't have a strong opinion, this is just a suggestion in case the name `index` sounds too generic and to prevent a potential conflict with a prop that might be named the same.

Reply:
Updated in 32ed2480b3004b1eea616cdcc2e7e9a333633580 
I stopped short of refactoring the prop in `grid-item.native.js` to avoid any regressions. It's probably low risk to rename the prop in the `Grid` component but I'd prefer to leave that as is and get this PR shipped.